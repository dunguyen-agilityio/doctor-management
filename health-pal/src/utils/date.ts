import dayjs from 'dayjs'

export const formatTime = (time: string, separator = '.') => {
  const splitTime = time.split(':')

  return `${splitTime.slice(0, 2).join(separator)} ${parseInt(splitTime[0]) > 12 ? 'PM' : 'AM'}`
}

export const getDefaultDate = () => {
  let now = dayjs()

  while ([0, 6].includes(now.day())) {
    if (![0, 6].includes(now.day())) break
    now = now.set('date', now.get('date') + 1)
  }

  return now
}
