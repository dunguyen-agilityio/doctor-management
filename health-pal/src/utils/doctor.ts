import { DoctorData, TDoctorCard } from '@app/models/doctor'

export const formatDoctor = ({
  clinic,
  specialty,
  users_permissions_user,
  ...rest
}: DoctorData): TDoctorCard => {
  const { avatar, ...user } = users_permissions_user

  return {
    ...user,
    ...rest,
    address: clinic.address,
    avatar: avatar?.url ?? '',
    specialty: specialty.name,
  }
}
