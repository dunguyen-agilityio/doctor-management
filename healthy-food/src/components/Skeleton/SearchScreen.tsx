import { StyleSheet, View } from 'react-native';

import { COLOR } from '@/theme';

import FilterSkeleton from './Filter';
import SearchInputSkeleton from './SearchInput';
import VerticalFoodListSkeleton from './VerticalFoodList';

const SearchScreenSkeleton = () => {
  return (
    <View style={styles.container}>
      <View style={styles.header}>
        <SearchInputSkeleton />
        <FilterSkeleton />
      </View>
      <VerticalFoodListSkeleton paddingTop={25} />
    </View>
  );
};

export default SearchScreenSkeleton;

const styles = StyleSheet.create({
  container: {
    paddingHorizontal: 20,
    flex: 1,
    paddingTop: 14,
    backgroundColor: COLOR.WHITE,
  },
  header: {
    gap: 16,
  },
});
