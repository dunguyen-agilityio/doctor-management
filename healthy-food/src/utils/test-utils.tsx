import {
  RenderOptions,
  render,
  renderHook,
} from '@testing-library/react-native';

import { NavigationContainer } from '@react-navigation/native';
import { QueryClient, QueryClientProvider } from '@tanstack/react-query';

const AllTheProviders = ({ children }: React.PropsWithChildren) => {
  const queryClient = new QueryClient({
    defaultOptions: { queries: { retry: false } },
  });

  return (
    <QueryClientProvider client={queryClient}>
      <NavigationContainer>{children}</NavigationContainer>
    </QueryClientProvider>
  );
};

const customRender = (ui: React.ReactElement, options?: RenderOptions) =>
  render(ui, { wrapper: AllTheProviders, ...options });

const customRenderHook: typeof renderHook = (renderCallback, options) =>
  renderHook(renderCallback, { wrapper: AllTheProviders, ...options });

export * from '@testing-library/react-native';

export { customRender as render, customRenderHook as renderHook };
