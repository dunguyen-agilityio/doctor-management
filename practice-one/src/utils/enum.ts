export const isInEnum = <T extends Record<string, string | number>>(
  obj: T,
  key: unknown = null,
): key is keyof T => {
  if (key == null) return false;
  return Object.keys(obj).includes(String(key));
};

export const isArrayEqual = <T>(arr1: T[], arr2: T[]) =>
  arr1.length == arr2.length && arr1.every((item) => arr2.includes(item));
