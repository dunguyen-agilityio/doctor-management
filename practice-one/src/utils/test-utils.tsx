import { RenderOptions, render } from '@testing-library/react-native';

import { NavigationContainer } from '@react-navigation/native';
import { QueryClient, QueryClientProvider } from '@tanstack/react-query';

const AllTheProviders = ({ children }: React.PropsWithChildren) => {
  const queryClient = new QueryClient({
    defaultOptions: { queries: { retry: false } },
  });

  return (
    <QueryClientProvider client={queryClient}>
      <NavigationContainer>{children}</NavigationContainer>
    </QueryClientProvider>
  );
};

const customRender = (ui: React.ReactElement, options?: RenderOptions) =>
  render(ui, { wrapper: AllTheProviders, ...options });

export * from '@testing-library/react-native';

export { customRender as render };
