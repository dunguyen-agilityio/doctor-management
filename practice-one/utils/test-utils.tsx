import { ReactElement } from 'react';

import { NavigationContainer } from '@react-navigation/native';

import { RenderOptions, render } from '@testing-library/react-native';

const AllTheProviders = ({ children }: { children: ReactElement }) => {
  return <NavigationContainer>{children}</NavigationContainer>;
};

const customRender = (ui: ReactElement, options?: RenderOptions) =>
  render(ui, { wrapper: AllTheProviders, ...options });

export * from '@testing-library/react-native';

export { customRender as render };
