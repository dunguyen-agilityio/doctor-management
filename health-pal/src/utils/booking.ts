import { BookingData } from '@/models/booking'

import { createDayjs } from './date'

export const formatBooking = ({ doctor, date, documentId, id, time, type }: BookingData) => {
  const { users_permissions_user, clinic, documentId: doctorDocId, id: doctorId } = doctor

  return {
    date: createDayjs(date),
    documentId,
    id,
    time,
    type,
    doctorName: users_permissions_user.name,
    doctorAvatar: users_permissions_user.avatar?.url,
    address: clinic.address,
    specialty: doctor.specialty.name,
    doctorId,
    doctorDocId,
  }
}
