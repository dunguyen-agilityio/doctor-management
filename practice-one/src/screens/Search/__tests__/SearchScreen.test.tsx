import { useRoute } from '@react-navigation/native';

import SearchScreen from '@/screens/Search';

import { render } from '@/utils/test-utils';

describe('SearchScreen', () => {
  it('match snapshot', () => {
    (useRoute as jest.Mock).mockReturnValue({});
    const tree = render(<SearchScreen />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('match snapshot', () => {
    (useRoute as jest.Mock).mockReturnValue({ params: { autoFocus: true } });
    const tree = render(<SearchScreen />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
