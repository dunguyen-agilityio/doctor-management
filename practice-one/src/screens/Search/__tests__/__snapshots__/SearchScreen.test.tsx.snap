// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchScreen match snapshot 1`] = `
<View
  style={
    {
      "backgroundColor": "#fff",
      "flex": 1,
    }
  }
>
  <View
    style={
      {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "space-between",
        "paddingHorizontal": 20,
        "width": "100%",
      }
    }
  >
    <Text
      style={
        {
          "color": "#5c5c5c",
          "fontFamily": "Manrope",
          "fontSize": 23,
          "fontWeight": "800",
          "letterSpacing": 0,
          "lineHeight": 32,
          "textTransform": undefined,
        }
      }
    >
      Want to eat
healthy Food?
    </Text>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      role="button"
      style={
        [
          {
            "alignItems": "center",
            "borderRadius": "100%",
            "justifyContent": "center",
          },
          {
            "backgroundColor": "#00000021",
            "width": 35,
          },
          {
            "height": 35,
          },
          undefined,
        ]
      }
      testID="question-icon"
    >
      <Image
        source={
          {
            "testUri": "../../../assets/icons/question.png",
          }
        }
      />
    </View>
  </View>
  <View
    style={
      {
        "backgroundColor": "#fff3f1",
        "borderRadius": 15,
        "height": 46,
        "marginHorizontal": 16,
        "marginTop": 14,
        "position": "relative",
      }
    }
    testID="search-input"
  >
    <View
      style={
        {
          "left": 21,
          "position": "absolute",
          "top": "50%",
          "transform": [
            {
              "translateY": -10,
            },
          ],
        }
      }
    >
      <Image
        source={
          {
            "testUri": "../../../assets/icons/icon-search.png",
          }
        }
      />
    </View>
    <TextInput
      autoFocus={false}
      defaultValue=""
      onChangeText={[Function]}
      placeholder="Search for healthy food"
      placeholderTextColor="#ff8473"
      style={
        {
          "color": "#000000",
          "fontSize": 13,
          "fontWeight": "400",
          "paddingHorizontal": 60,
          "paddingVertical": 14,
        }
      }
    />
  </View>
  <View
    style={
      {
        "height": 26,
        "marginTop": 16,
      }
    }
  >
    <RCTScrollView
      collapsable={false}
      contentContainerStyle={
        {
          "gap": 16,
        }
      }
      handlerTag={1}
      handlerType="NativeViewGestureHandler"
      horizontal={true}
      onGestureHandlerEvent={[Function]}
      onGestureHandlerStateChange={[Function]}
      pagingEnabled={true}
      showsHorizontalScrollIndicator={false}
      waitFor={
        [
          {
            "current": null,
          },
        ]
      }
    >
      <View>
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "alignItems": "center",
              "backgroundColor": "#ececec",
              "borderRadius": 8,
              "justifyContent": "center",
              "marginLeft": 16,
              "marginRight": 0,
              "opacity": 1,
              "paddingHorizontal": 10,
              "paddingVertical": 4,
            }
          }
          testID="category-item"
        >
          <Text
            style={
              {
                "color": "#5c5c5c",
                "fontFamily": "Manrope",
                "fontSize": 13,
                "fontWeight": "400",
                "letterSpacing": 0,
                "lineHeight": 18,
                "textTransform": undefined,
              }
            }
          >
            Breakfast
          </Text>
        </View>
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "alignItems": "center",
              "backgroundColor": "#ececec",
              "borderRadius": 8,
              "justifyContent": "center",
              "marginLeft": 0,
              "marginRight": 0,
              "opacity": 1,
              "paddingHorizontal": 10,
              "paddingVertical": 4,
            }
          }
          testID="category-item"
        >
          <Text
            style={
              {
                "color": "#5c5c5c",
                "fontFamily": "Manrope",
                "fontSize": 13,
                "fontWeight": "400",
                "letterSpacing": 0,
                "lineHeight": 18,
                "textTransform": undefined,
              }
            }
          >
            Dinner
          </Text>
        </View>
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "alignItems": "center",
              "backgroundColor": "#ececec",
              "borderRadius": 8,
              "justifyContent": "center",
              "marginLeft": 0,
              "marginRight": 0,
              "opacity": 1,
              "paddingHorizontal": 10,
              "paddingVertical": 4,
            }
          }
          testID="category-item"
        >
          <Text
            style={
              {
                "color": "#5c5c5c",
                "fontFamily": "Manrope",
                "fontSize": 13,
                "fontWeight": "400",
                "letterSpacing": 0,
                "lineHeight": 18,
                "textTransform": undefined,
              }
            }
          >
            Dry fruits
          </Text>
        </View>
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "alignItems": "center",
              "backgroundColor": "#ececec",
              "borderRadius": 8,
              "justifyContent": "center",
              "marginLeft": 0,
              "marginRight": 0,
              "opacity": 1,
              "paddingHorizontal": 10,
              "paddingVertical": 4,
            }
          }
          testID="category-item"
        >
          <Text
            style={
              {
                "color": "#5c5c5c",
                "fontFamily": "Manrope",
                "fontSize": 13,
                "fontWeight": "400",
                "letterSpacing": 0,
                "lineHeight": 18,
                "textTransform": undefined,
              }
            }
          >
            Lunch
          </Text>
        </View>
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "alignItems": "center",
              "backgroundColor": "#ececec",
              "borderRadius": 8,
              "justifyContent": "center",
              "marginLeft": 0,
              "marginRight": 0,
              "opacity": 1,
              "paddingHorizontal": 10,
              "paddingVertical": 4,
            }
          }
          testID="category-item"
        >
          <Text
            style={
              {
                "color": "#5c5c5c",
                "fontFamily": "Manrope",
                "fontSize": 13,
                "fontWeight": "400",
                "letterSpacing": 0,
                "lineHeight": 18,
                "textTransform": undefined,
              }
            }
          >
            Vegetables
          </Text>
        </View>
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "alignItems": "center",
              "backgroundColor": "#ececec",
              "borderRadius": 8,
              "justifyContent": "center",
              "marginLeft": 0,
              "marginRight": 0,
              "opacity": 1,
              "paddingHorizontal": 10,
              "paddingVertical": 4,
            }
          }
          testID="category-item"
        >
          <Text
            style={
              {
                "color": "#5c5c5c",
                "fontFamily": "Manrope",
                "fontSize": 13,
                "fontWeight": "400",
                "letterSpacing": 0,
                "lineHeight": 18,
                "textTransform": undefined,
              }
            }
          >
            Greenish
          </Text>
        </View>
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "alignItems": "center",
              "backgroundColor": "#ececec",
              "borderRadius": 8,
              "justifyContent": "center",
              "marginLeft": 0,
              "marginRight": 0,
              "opacity": 1,
              "paddingHorizontal": 10,
              "paddingVertical": 4,
            }
          }
          testID="category-item"
        >
          <Text
            style={
              {
                "color": "#5c5c5c",
                "fontFamily": "Manrope",
                "fontSize": 13,
                "fontWeight": "400",
                "letterSpacing": 0,
                "lineHeight": 18,
                "textTransform": undefined,
              }
            }
          >
            Non-Veg
          </Text>
        </View>
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "alignItems": "center",
              "backgroundColor": "#ececec",
              "borderRadius": 8,
              "justifyContent": "center",
              "marginLeft": 0,
              "marginRight": 16,
              "opacity": 1,
              "paddingHorizontal": 10,
              "paddingVertical": 4,
            }
          }
          testID="category-item"
        >
          <Text
            style={
              {
                "color": "#5c5c5c",
                "fontFamily": "Manrope",
                "fontSize": 13,
                "fontWeight": "400",
                "letterSpacing": 0,
                "lineHeight": 18,
                "textTransform": undefined,
              }
            }
          >
            Fast Food
          </Text>
        </View>
      </View>
    </RCTScrollView>
  </View>
  <View
    style={
      [
        {
          "alignItems": "center",
          "height": "100%",
          "justifyContent": "center",
          "position": "absolute",
          "width": "100%",
          "zIndex": 10,
        },
        {
          "marginTop": undefined,
        },
      ]
    }
    testID="loading-indicator"
  >
    <View
      collapsable={false}
      style={
        {
          "transform": [
            {
              "rotate": "0deg",
            },
          ],
        }
      }
    >
      <Image
        source={
          {
            "testUri": "../../../assets/images/logo.png",
          }
        }
        style={
          {
            "height": 122,
            "width": 122,
          }
        }
      />
    </View>
  </View>
</View>
`;

exports[`SearchScreen match snapshot 2`] = `
<View
  style={
    {
      "backgroundColor": "#fff",
      "flex": 1,
    }
  }
>
  <View
    style={
      {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "space-between",
        "paddingHorizontal": 20,
        "width": "100%",
      }
    }
  >
    <Text
      style={
        {
          "color": "#5c5c5c",
          "fontFamily": "Manrope",
          "fontSize": 23,
          "fontWeight": "800",
          "letterSpacing": 0,
          "lineHeight": 32,
          "textTransform": undefined,
        }
      }
    >
      Want to eat
healthy Food?
    </Text>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      role="button"
      style={
        [
          {
            "alignItems": "center",
            "borderRadius": "100%",
            "justifyContent": "center",
          },
          {
            "backgroundColor": "#00000021",
            "width": 35,
          },
          {
            "height": 35,
          },
          undefined,
        ]
      }
      testID="question-icon"
    >
      <Image
        source={
          {
            "testUri": "../../../assets/icons/question.png",
          }
        }
      />
    </View>
  </View>
  <View
    style={
      {
        "backgroundColor": "#fff3f1",
        "borderRadius": 15,
        "height": 46,
        "marginHorizontal": 16,
        "marginTop": 14,
        "position": "relative",
      }
    }
    testID="search-input"
  >
    <View
      style={
        {
          "left": 21,
          "position": "absolute",
          "top": "50%",
          "transform": [
            {
              "translateY": -10,
            },
          ],
        }
      }
    >
      <Image
        source={
          {
            "testUri": "../../../assets/icons/icon-search.png",
          }
        }
      />
    </View>
    <TextInput
      autoFocus={true}
      defaultValue=""
      onChangeText={[Function]}
      placeholder="Search for healthy food"
      placeholderTextColor="#ff8473"
      style={
        {
          "color": "#000000",
          "fontSize": 13,
          "fontWeight": "400",
          "paddingHorizontal": 60,
          "paddingVertical": 14,
        }
      }
    />
  </View>
  <View
    style={
      {
        "height": 26,
        "marginTop": 16,
      }
    }
  >
    <RCTScrollView
      collapsable={false}
      contentContainerStyle={
        {
          "gap": 16,
        }
      }
      handlerTag={2}
      handlerType="NativeViewGestureHandler"
      horizontal={true}
      onGestureHandlerEvent={[Function]}
      onGestureHandlerStateChange={[Function]}
      pagingEnabled={true}
      showsHorizontalScrollIndicator={false}
      waitFor={
        [
          {
            "current": null,
          },
        ]
      }
    >
      <View>
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "alignItems": "center",
              "backgroundColor": "#ececec",
              "borderRadius": 8,
              "justifyContent": "center",
              "marginLeft": 16,
              "marginRight": 0,
              "opacity": 1,
              "paddingHorizontal": 10,
              "paddingVertical": 4,
            }
          }
          testID="category-item"
        >
          <Text
            style={
              {
                "color": "#5c5c5c",
                "fontFamily": "Manrope",
                "fontSize": 13,
                "fontWeight": "400",
                "letterSpacing": 0,
                "lineHeight": 18,
                "textTransform": undefined,
              }
            }
          >
            Breakfast
          </Text>
        </View>
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "alignItems": "center",
              "backgroundColor": "#ececec",
              "borderRadius": 8,
              "justifyContent": "center",
              "marginLeft": 0,
              "marginRight": 0,
              "opacity": 1,
              "paddingHorizontal": 10,
              "paddingVertical": 4,
            }
          }
          testID="category-item"
        >
          <Text
            style={
              {
                "color": "#5c5c5c",
                "fontFamily": "Manrope",
                "fontSize": 13,
                "fontWeight": "400",
                "letterSpacing": 0,
                "lineHeight": 18,
                "textTransform": undefined,
              }
            }
          >
            Dinner
          </Text>
        </View>
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "alignItems": "center",
              "backgroundColor": "#ececec",
              "borderRadius": 8,
              "justifyContent": "center",
              "marginLeft": 0,
              "marginRight": 0,
              "opacity": 1,
              "paddingHorizontal": 10,
              "paddingVertical": 4,
            }
          }
          testID="category-item"
        >
          <Text
            style={
              {
                "color": "#5c5c5c",
                "fontFamily": "Manrope",
                "fontSize": 13,
                "fontWeight": "400",
                "letterSpacing": 0,
                "lineHeight": 18,
                "textTransform": undefined,
              }
            }
          >
            Dry fruits
          </Text>
        </View>
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "alignItems": "center",
              "backgroundColor": "#ececec",
              "borderRadius": 8,
              "justifyContent": "center",
              "marginLeft": 0,
              "marginRight": 0,
              "opacity": 1,
              "paddingHorizontal": 10,
              "paddingVertical": 4,
            }
          }
          testID="category-item"
        >
          <Text
            style={
              {
                "color": "#5c5c5c",
                "fontFamily": "Manrope",
                "fontSize": 13,
                "fontWeight": "400",
                "letterSpacing": 0,
                "lineHeight": 18,
                "textTransform": undefined,
              }
            }
          >
            Lunch
          </Text>
        </View>
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "alignItems": "center",
              "backgroundColor": "#ececec",
              "borderRadius": 8,
              "justifyContent": "center",
              "marginLeft": 0,
              "marginRight": 0,
              "opacity": 1,
              "paddingHorizontal": 10,
              "paddingVertical": 4,
            }
          }
          testID="category-item"
        >
          <Text
            style={
              {
                "color": "#5c5c5c",
                "fontFamily": "Manrope",
                "fontSize": 13,
                "fontWeight": "400",
                "letterSpacing": 0,
                "lineHeight": 18,
                "textTransform": undefined,
              }
            }
          >
            Vegetables
          </Text>
        </View>
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "alignItems": "center",
              "backgroundColor": "#ececec",
              "borderRadius": 8,
              "justifyContent": "center",
              "marginLeft": 0,
              "marginRight": 0,
              "opacity": 1,
              "paddingHorizontal": 10,
              "paddingVertical": 4,
            }
          }
          testID="category-item"
        >
          <Text
            style={
              {
                "color": "#5c5c5c",
                "fontFamily": "Manrope",
                "fontSize": 13,
                "fontWeight": "400",
                "letterSpacing": 0,
                "lineHeight": 18,
                "textTransform": undefined,
              }
            }
          >
            Greenish
          </Text>
        </View>
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "alignItems": "center",
              "backgroundColor": "#ececec",
              "borderRadius": 8,
              "justifyContent": "center",
              "marginLeft": 0,
              "marginRight": 0,
              "opacity": 1,
              "paddingHorizontal": 10,
              "paddingVertical": 4,
            }
          }
          testID="category-item"
        >
          <Text
            style={
              {
                "color": "#5c5c5c",
                "fontFamily": "Manrope",
                "fontSize": 13,
                "fontWeight": "400",
                "letterSpacing": 0,
                "lineHeight": 18,
                "textTransform": undefined,
              }
            }
          >
            Non-Veg
          </Text>
        </View>
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "alignItems": "center",
              "backgroundColor": "#ececec",
              "borderRadius": 8,
              "justifyContent": "center",
              "marginLeft": 0,
              "marginRight": 16,
              "opacity": 1,
              "paddingHorizontal": 10,
              "paddingVertical": 4,
            }
          }
          testID="category-item"
        >
          <Text
            style={
              {
                "color": "#5c5c5c",
                "fontFamily": "Manrope",
                "fontSize": 13,
                "fontWeight": "400",
                "letterSpacing": 0,
                "lineHeight": 18,
                "textTransform": undefined,
              }
            }
          >
            Fast Food
          </Text>
        </View>
      </View>
    </RCTScrollView>
  </View>
  <View
    style={
      [
        {
          "alignItems": "center",
          "height": "100%",
          "justifyContent": "center",
          "position": "absolute",
          "width": "100%",
          "zIndex": 10,
        },
        {
          "marginTop": undefined,
        },
      ]
    }
    testID="loading-indicator"
  >
    <View
      collapsable={false}
      style={
        {
          "transform": [
            {
              "rotate": "0deg",
            },
          ],
        }
      }
    >
      <Image
        source={
          {
            "testUri": "../../../assets/images/logo.png",
          }
        }
        style={
          {
            "height": 122,
            "width": 122,
          }
        }
      />
    </View>
  </View>
</View>
`;
